runtime/src/system_instruction_processor.rs

```diff
@@ -50,11 +92,18 @@ fn create_system_account(
         return Err(SystemError::ResultWithNegativeLamports.into());
     }
 
-    assign_account_to_program(to, program_id)?;
```

Comment:
Not calling `assign_account_to_program` anymore for normal accounts starts to skip [this existing check for account.signer_key](https://github.com/solana-labs/solana/pull/7390/files#diff-f06d4cdb8c607b22d0504416e326b5cfL65). Is that OK?

Reply:
that check is already present in `create_account()` and is not enforced by `create_account_with_seed()`

this change was intentional

Reply:
Ok, thanks for clarifying!